File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum=INT_MIN,msf=0;
        for(int i: nums){
            if(msf<i && msf<0){
                msf=i;
            }
            else msf+=i;
            sum=max(sum,msf);
        }
        return sum;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int sum =Integer.MIN_VALUE,msf=0;
        for(int i=0;i<nums.length;i++){
            if(msf<nums[i] && msf<0){
                msf=nums[i];
            }
            else msf+=nums[i];
            sum=Math.max(sum,msf);
        }
        return sum;
    }
}